.github/workflows/ci.yml

```
@@ -20,39 +20,59 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-16.04
+    runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
         include:
           - framework: tensorflow
-            python: 3.6
-            tensorflow: 1.15.3
+            python: 3.7
+            tensorflow: 1.15.4
             tf_version: v1
             keras: 2.2.5
           - framework: tensorflow
-            python: 3.6
+            python: 3.7
             tensorflow: 2.2.0
             tf_version: v2
             keras: 2.3.1
+          - framework: tensorflow
+            python: 3.7
+            tensorflow: 2.3.1
+            tf_version: v2
+            keras: 2.4.3
+          - framework: tensorflow
+            python: 3.8
+            tensorflow: 2.4.0rc1
+            tf_version: v2
+            keras: 2.4.3
           - framework: keras
             python: 3.7
-            tensorflow: 2.2.0
+            tensorflow: 2.2.1
             keras: 2.3.1
           - framework: kerastf
             python: 3.7
-            tensorflow: 2.2.0
-            keras: 2.3.1
+            tensorflow: 2.3.1
+            keras: 2.4.3
           - framework: pytorch
             python: 3.7
           - framework: scikitlearn
+            scikit-learn: 0.22.2
             python: 3.7
+          - framework: scikitlearn
+            scikit-learn: 0.23.2
+            python: 3.8
           - framework: mxnet
             python: 3.7
           - framework: legacy
             python: 3.7
-            tensorflow: 2.2.0
+            tensorflow: 2.2.1
             keras: 2.3.1
+            scikit-learn: 0.22.2
+          - framework: legacy
+            python: 3.8
+            tensorflow: 2.3.1
+            keras: 2.4.3
+            scikit-learn: 0.23.2
 
     name: Run ${{ matrix.framework }} ${{ matrix.tf_version }} Tests
```

After adding short names to each variant, this line can be changed.
```suggestion
    name: Run ${{ matrix.name }} Tests
```